app/src/main/java/com/example/trainingintive/viewmodels/SplashViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.trainingintive.viewmodels
+
+import androidx.lifecycle.ViewModel
+import com.example.trainingintive.Navigator
+import io.reactivex.rxjava3.android.schedulers.AndroidSchedulers
+import io.reactivex.rxjava3.core.Single
+import io.reactivex.rxjava3.disposables.Disposable
+import io.reactivex.rxjava3.schedulers.Schedulers
+import java.util.concurrent.TimeUnit
+
+class SplashViewModel : ViewModel() {
+
+    lateinit var navigator: Navigator
+    lateinit var activityStatus: ActivityStatus
+    var event: Event
+    val downloadData: Disposable
+
+    init {
+        event = Event.Downloading
+        downloadData = Single.fromCallable({ downloadData() })
+            .delay(5, TimeUnit.SECONDS)
+            .subscribeOn(AndroidSchedulers.mainThread())
+            .observeOn(Schedulers.io())
+            .subscribe(
+                {
+                    event = Event.OpenActivity
+                    start()
+                },
+                { event = Event.Error }
+            )
+    }
+
+    fun downloadData() {}
+
+    fun start() {
+        if (activityStatus is ActivityStatus.Active && event is Event.OpenActivity) {
```

Comment:
ViewModel shouldn't check view state (view - activity in this case), view should observe data from ViewModel.
Sounds like a job for a [LiveData](https://developer.android.com/topic/libraries/architecture/livedata) 
As a proposal: 
- you should have possibility to send events to navigator,
- navigator should handle each event separately (start activity, show dialog, use navcontroller),

Reply:
In summary: ActivityStatus should be dropped